File: my/src/main/java/com/dhc/flyabbit/my/MyFragment.java
package com.dhc.flyabbit.my;

import android.os.Bundle;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.dhc.library.utils.delegate.OnBackToFirstListener;
import com.dhc.library.base.BaseActivity;
import com.dhc.library.base.BaseFragment;
import com.dhc.library.base.XDaggerFragment;
import com.dhc.library.data.bean.ToolBarOptions;
import com.dhc.library.base.WebViewCommonFragment;

/**
 * 创建者：邓浩宸
 * 时间 ：2017/6/30 0030 上午 10:23
 * 描述 ：TODO 请描述该类职责
 */

public class MyFragment extends XDaggerFragment implements View.OnClickListener {

    protected OnBackToFirstListener _mBackToFirstListener;
    protected  ImageView mIvLogo;
    private TextView mGithub;
    private  TextView mAuthor;
    public static MyFragment newInstance() {
        MyFragment fragment = new MyFragment();
        return fragment;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_my_main;
    }

    @Override
    protected void initEventAndData(View view) {
        BaseFragment baseFragment = (BaseFragment) getParentFragment();
        if (baseFragment instanceof OnBackToFirstListener) {
            _mBackToFirstListener = (OnBackToFirstListener) baseFragment;
        }
        initTitle();
        initView();
        initListener();
    }
    private void initTitle() {
        ToolBarOptions options = new ToolBarOptions()
                .isNeedNavigate(false).titleString("我的");
        setToolBar(R.id.toolbar, options);
    }
    private void initView() {
        mIvLogo = $(R.id.iv_logo);
        mGithub = $(R.id.tv_github);
        mAuthor = $(R.id.tv_author);
        mIvLogo.setColorFilter(ContextCompat.getColor(_mActivity, R.color.clr_6385a7));
    }

    private void initListener() {
        mGithub.setOnClickListener(this);
        mAuthor.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if(v.getId()==R.id.tv_github){
            ((BaseActivity) _mActivity).start(WebViewCommonFragment.newInstance("Github",
                   "https://github.com/chengzichen/Flyabbit", "", ""));
        }else if (v.getId()==R.id.tv_author){
            ((BaseActivity) _mActivity).start(WebViewCommonFragment.newInstance("个人博客",
                    "http://blog.csdn.net/chengzichen_/article/details/77659318", "", ""));
        }
    }

    @Override
    public void initInject(Bundle savedInstanceState) {
    }
    


    /**
     * 处理回退事件
     *
     * @return
     */
    @Override
    public boolean onBackPressedSupport() {
        if (getChildFragmentManager().getBackStackEntryCount() > 1) {
            popChild();
        } else {
            if (_mBackToFirstListener != null) {
                _mBackToFirstListener.onBackToFirstFragment();
            } else {
                _mActivity.finish();
            }
        }
        return true;
    }


}

File: my/src/main/java/com/dhc/flyabbit/my/MyFragment.kt
package com.dhc.flyabbit.my

import android.os.Bundle
import androidx.core.content.ContextCompat
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.alibaba.android.arouter.facade.annotation.Route
import com.dhc.lib.widget.util.ToolbarUtil
import com.dhc.library.base.BaseActivity
import com.dhc.library.base.BaseFragment
import com.dhc.businesscomponent.base.WebViewCommonFragment
import com.dhc.businesscomponent.framework.OnBackToFirstListener
import com.dhc.library.base.XDaggerFragment
import com.dhc.lib.widget.bean.ToolBarOptions
import com.dhc.library.framework.IBasePresenter
import com.dhc.library.framework.IBaseView

/**
 * 创建者：邓浩宸
 * 时间 ：2017/6/30 0030 上午 10:23
 * 描述 ：TODO 请描述该类职责
 */
@Route(path = "/my/MyFragment")
class MyFragment : XDaggerFragment<IBasePresenter<IBaseView>,IBaseView>(), View.OnClickListener {

    protected var _mBackToFirstListener: OnBackToFirstListener? = null
    protected lateinit var mIvLogo: ImageView
    private var mGithub: TextView? = null
    private var mAuthor: TextView? = null


    override fun initEventAndData(savedInstanceState: Bundle?) {
        val baseFragment = parentFragment as BaseFragment?
        if (baseFragment is OnBackToFirstListener) {
            _mBackToFirstListener = baseFragment
        }
        initTitle()
        initView()
        initListener()
    }

    private fun initTitle() {
        val options = ToolBarOptions()
                .isNeedNavigate(false).titleString("我的")
        ToolbarUtil().setToolBar(_mActivity, `$`<View>(R.id.toolbar) as androidx.appcompat.widget.Toolbar, options, false)
        //        setToolBar(R.id.toolbar, options);
    }

    private fun initView() {
        mIvLogo = `$`(R.id.iv_logo)
        mGithub = `$`(R.id.tv_github)
        mAuthor = `$`(R.id.tv_author)
        mIvLogo.setColorFilter(ContextCompat.getColor(_mActivity, R.color.clr_6385a7))
    }

    private fun initListener() {
        mGithub!!.setOnClickListener(this)
        mAuthor!!.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        if (v.id == R.id.tv_github) {
            (_mActivity as BaseActivity).start(WebViewCommonFragment.newInstance("Github",
                    "https://github.com/chengzichen/Flyabbit", "", "", true))
        } else if (v.id == R.id.tv_author) {
            (_mActivity as BaseActivity).start(WebViewCommonFragment.newInstance("个人博客",
                    "http://blog.csdn.net/chengzichen_/article/details/77659318", "", "", true))
        }
    }

    override fun initInject(savedInstanceState: Bundle?) {

    }


    /**
     * 处理回退事件
     *
     * @return
     */
    override fun onBackPressedSupport(): Boolean {
        if (childFragmentManager.backStackEntryCount > 1) {
            popChild()
        } else {
            if (_mBackToFirstListener != null) {
                _mBackToFirstListener!!.onBackToFirstFragment()
            } else {
                _mActivity.finish()
            }
        }
        return true
    }

    override val layoutId: Int
        get() = R.layout.fragment_my_main

}
